app/models/katello/module_stream.rb

```diff
@@ -50,5 +50,15 @@ def create_profiles(profiles)
         end
       end
     end
+
+    def self.available_for_hosts(hosts)
+      joins("INNER JOIN #{Katello::RepositoryModuleStream.table_name} on \
+        #{Katello::RepositoryModuleStream.table_name}.module_stream_id = #{Katello::ModuleStream.table_name}.id").
+      joins("INNER JOIN #{Katello::ContentFacetRepository.table_name} on \
+        #{Katello::RepositoryModuleStream.table_name}.repository_id = #{Katello::ContentFacetRepository.table_name}.id").
+      joins("INNER JOIN #{Katello::Host::ContentFacet.table_name} on \
+        #{Katello::Host::ContentFacet.table_name}.id = #{Katello::ContentFacetRepository.table_name}.content_facet_id").
+      where("#{Katello::Host::ContentFacet.table_name}.host_id" => hosts).distinct
+    end
```

Comment:
You have a couple options in AR for this (assuming I'm understanding the query correctly)
```ruby
Katello::ModuleStream.joins(repositories: :content_facets).where('katello_content_facets.host_id' => [4])
# OR
Katello::ModuleStream.joins(repositories: :content_facets).merge( Katello::Host::ContentFacet.where(host_id: [4]) )
```